packages/flutter/test/widgets/nested_scroll_view_async_test.dart

```diff
@@ -13,41 +13,41 @@ void main() {
     WidgetsBinding.instance.resetEpoch();
   });
 
-  test('NestedScrollView can build sccessfully if mark dirty during warm up frame', () {
+  test('NestedScrollView can rebuild twice before layout', () {
     final FakeAsync fakeAsync = FakeAsync();
     fakeAsync.run((FakeAsync async) {
-      runApp(
-        MaterialApp(
-          home: Material(
-            child: DefaultTabController(
-              length: 1,
-              child: NestedScrollView(
-                dragStartBehavior: DragStartBehavior.down,
-                headerSliverBuilder: (BuildContext context, bool innerBoxIsScrolled) {
-                  return <Widget>[
-                    const SliverPersistentHeader(
-                      delegate: TestHeader(),
-                    ),
-                  ];
-                },
-                body: SingleChildScrollView(
-                  dragStartBehavior: DragStartBehavior.down,
-                  child: Container(
-                    height: 1000.0,
-                    child: const Placeholder(),
+      final Widget app = MaterialApp(
+        home: Material(
+          child: DefaultTabController(
+            length: 1,
+            child: NestedScrollView(
+              dragStartBehavior: DragStartBehavior.down,
+              headerSliverBuilder: (BuildContext context, bool innerBoxIsScrolled) {
+                return <Widget>[
+                  const SliverPersistentHeader(
+                    delegate: TestHeader(),
                   ),
+                ];
+              },
+              body: SingleChildScrollView(
+                dragStartBehavior: DragStartBehavior.down,
+                child: Container(
+                  height: 1000.0,
+                  child: const Placeholder(),
                 ),
               ),
             ),
           ),
         ),
       );
-      // Marks element as dirty right before the first draw frame is called.
-      // This can happen when engine flush user setting.
+      // Initiates the first build.
```

Comment:
since it is no longer possible to dirty thing in the middle of warm up frame, I have to change the way to reproduce double builds. if we are ok to remove the null check in nestedscrollview, we can remove this test as well